### ISSUE TITLE
CommentsIndentation Check - False Positive at End-of-Method

### ISSUE THREAD

**Issue by gkeighren** [checkstyle/checkstyle#2998]
**Details**
```
/var/tmp $ /cygdrive/c/Program\ Files/Java/jdk1.8.0_66/bin/javac.exe Test.java

/var/tmp $ cat Test.java
public class Test {
    private final Object anyObject = null;

    public void method1() {
        anyObject.toString();
        // OK
    }

    public void method2() {
        // Single-line comment
        anyObject.toString();
        // OK
    }

    public void method3() {
        // Multi-line block comment
        // Here is the second line
        anyObject.toString();
        // FALSE-POSITIVE HERE
    }

    public void method4() throws InterruptedException {
        // Multi-line block comment
        // Here is the second line
        anyObject.wait();
        // FALSE-POSITIVE HERE
    }
}

/var/tmp $ cat config.xml
<!DOCTYPE module PUBLIC
"-//Puppy Crawl//DTD Check Configuration 1.3//EN"
"http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="CommentsIndentation"/>
  </module>
</module>

/var/tmp $ java -jar checkstyle-6.16-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /var/tmp/Test.java:20: Comment has incorrect indentation level 1, expected is 19, indentation should be the same level as line 19. [CommentsIndentation]
[ERROR] /var/tmp/Test.java:28: Comment has incorrect indentation level 1, expected is 15, indentation should be the same level as line 27. [CommentsIndentation]
Audit done.
Checkstyle ends with 2 errors.
```

**Expected Outcome**
No violations.

**Comment by romani** [comment]
@MEZk , please put a comment on this request.

**Comment by MEZk** [comment]
@gkeighren
Thanks for the report.

@romani 
Yes, it is a bug. Please assign.

**Comment by romani** [comment]
@MEZk , you have enough assignments, please do them first.

**Comment by Vladlis** [comment]
A fix is provided in #3123 
However there are other false-positives, which should also be fixed:
```
vlad@vlad-N61PC-M2S ~/cli $ cat Test.java
public class Test {
    void foo1() {
        if (true) System.out.println(
                "some string"); // line 4
        // violation - line 5
    }

    void foo2() {
        foo1();
        /*
         line 11 */
        // violation - line 12
    }

    void foo3() {
        String[] array = { // line 16
            // violation - line 17
        };
    }
}

vlad@vlad-N61PC-M2S ~/cli $ cat my_check.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">


<module name = "Checker">
    <module name="TreeWalker">
         <module name="CommentsIndentation">
         </module>
    </module>
</module>
vlad@vlad-N61PC-M2S ~/cli $ java -jar checkstyle-6.17-all.jar com.puppycrawl.tools.checkstyle.Main -c my_check.xml Test.java
Starting audit...
[ERROR] /home/vlad/cli/Test.java:5: Comment has incorrect indentation level 8, expected is 16, indentation should be the same level as line 4. [CommentsIndentation]
[ERROR] /home/vlad/cli/Test.java:12: Comment has incorrect indentation level 8, expected is 17, indentation should be the same level as line 11. [CommentsIndentation]
[ERROR] /home/vlad/cli/Test.java:17: Comment has incorrect indentation level 12, expected is 25, indentation should be the same level as line 16. [CommentsIndentation]
Audit done.
Checkstyle ends with 3 errors.
```
Expected - no errors.

**Comment by romani** [comment]
additional false-positives:

1)
openjdk7/src/share/classes/sun/management/snmp/jvminstr/JvmMemGCTableMetaImpl.java.html#L124
CommentsIndentation 	Comment has incorrect indentation level 16, expected is 22, indentation should be the same level as line 123. 	124
```
                 if (dbg) log.debug("GCTableFilter",
                       ((MemoryManagerMXBean)value).getName() +
                       " is not a  GarbageCollectorMXBean: " +
                       value.getClass().getName());
                 // skip to next index... // violation!!!
``` 	

2)
openjdk7/src/share/classes/sun/security/pkcs11/P11KeyPairGenerator.java.html#L242
CommentsIndentation 	Comment has incorrect indentation level 16, expected is 12, indentation should be the same level as line 241. 	242
```
             privateKeyTemplate = new CK_ATTRIBUTE[] {
                 // empty // violation !!!
            };
```

3)
openjdk7/src/share/classes/sun/security/tools/policytool/PolicyTool.java.html#L4127
violation become a bit more consistent as expected indentation is printed correctly. But this whole violation is false positive - there should be no violation.

- # 	warning 	CommentsIndentation 	Comment has incorrect indentation level 16, expected is 24, indentation should be the same level as line 4126. 	4127 	
+ # 	warning 	CommentsIndentation 	Comment has incorrect indentation level 16, expected is 8, indentation should be the same level as line 4126. 	4127 	

```
class SubjDelegPerm extends Perm {
     public SubjDelegPerm() {
     super("SubjectDelegationPermission",
         "javax.management.remote.SubjectDelegationPermission",
         new String[]    {
                 // allow user input // violation !!!
                 },
         null);
     }
}
```

another example of the same problem (no violation is expected):
openjdk7/test/sun/nio/cs/TestEUC_TW.java.html#L284
```
     // The first byte is the length of malformed bytes
     static byte[][] malformed = {
         //{5, (byte)0xF8, (byte)0x80, (byte)0x80, (byte)0x9F, (byte)0x80, (byte)0xC0 },
     };
```

**Comment by Vladlis** [comment]
@romani 
these aditional false-positives are covered in my example:
1) is covered by `foo1()`
2) and 3) are covered by `foo3()`

**Comment by romani** [comment]
yes, I see .
Please create 3 different issues for this.  it will help us to test regressions easily.
I will close this issue.

**Comment by Vladlis** [comment]
@romani 
Done
#3125 
#3126 
#3127

**Comment by romani** [comment]
this issue is closed
